uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -0,0 +1,1208 @@
+            @Override
+            protected void assertAliasIdpAfterDeletion(final String aliasId, final String aliasZid) throws Exception {
+                // if the alias feature is enabled, the alias should also be removed
+                assertIdpDoesNotExist(aliasId, aliasZid);
+            }
+        }
+
+        @Nested
+        class AliasFeatureDisabled extends DeleteBase {
+            public AliasFeatureDisabled() {
+                super(false);
+            }
+
+            @Override
+            protected void assertAliasIdpAfterDeletion(final String aliasId, final String aliasZid) throws Exception {
+                // if the alias feature is disabled, only the reference should be removed from the alias IdP
+                assertReferenceWasRemovedFromAlias(aliasId, aliasZid);
+            }
+        }
+
+
+        private MvcResult deleteIdpAndReturnResult(final IdentityZone zone, final String id) throws Exception {
+            final String accessTokenForZone1 = getAccessTokenForZone(zone.getId());
+            final MockHttpServletRequestBuilder deleteRequestBuilder = delete("/identity-providers/" + id)
+                    .header("Authorization", "Bearer " + accessTokenForZone1)
+                    .header(IdentityZoneSwitchingFilter.HEADER, zone.getId());
+            return mockMvc.perform(deleteRequestBuilder).andReturn();
+        }
+
+        private void assertIdpDoesNotExist(final String id, final String zoneId) throws Exception {
+            final Optional<IdentityProvider<?>> idp = readIdpFromZoneIfExists(zoneId, id);
+            assertThat(idp).isNotPresent();
+        }
+    }
+
+    private void assertReferenceWasRemovedFromAlias(final String aliasId, final String aliasZid) throws Exception {
+        final Optional<IdentityProvider<?>> aliasIdpAfterDeletion = readIdpFromZoneIfExists(aliasZid, aliasId);
+        assertThat(aliasIdpAfterDeletion).isPresent();
+        assertThat(aliasIdpAfterDeletion.get().getAliasId()).isBlank();
+        assertThat(aliasIdpAfterDeletion.get().getAliasZid()).isBlank();
+    }
+
+    private void assertIdpReferencesOtherIdp(final IdentityProvider<?> idp, final IdentityProvider<?> referencedIdp) {
+        assertThat(idp).isNotNull();
+        assertThat(referencedIdp).isNotNull();
+        assertThat(referencedIdp.getId()).isNotBlank().isEqualTo(idp.getAliasId());
+        assertThat(referencedIdp.getIdentityZoneId()).isNotBlank().isEqualTo(idp.getAliasZid());
+    }
+
+    private void assertOtherPropertiesAreEqual(final IdentityProvider<?> idp, final IdentityProvider<?> aliasIdp) {
+        // the configs should be identical
+        final OIDCIdentityProviderDefinition originalIdpConfig = (OIDCIdentityProviderDefinition) idp.getConfig();
+        final OIDCIdentityProviderDefinition aliasIdpConfig = (OIDCIdentityProviderDefinition) aliasIdp.getConfig();
+        assertThat(aliasIdpConfig).isEqualTo(originalIdpConfig);
+
+        // check if remaining properties are equal
```

Comment:
Are these 3 remaining properties the only relevant properties, if so, should this test be called assertImportantPropertiesAreEqual or Relevant, Primary, or Significant Properties? There are many other properties like LastModified, version, identityZoneID and such that are not checked.

Reply:
I checked all properties again and found out that I missed to check for the `active` field. The only properties that won't be equal are the two timestamps `created` and `lastmodified` as well as the `version` field, which is fine IMO.

I added the missing check as well as a comment about the other properties in f12dc96.